catroid/src/main/java/org/catrobat/catroid/ui/ScratchConverterActivity.java

```diff
@@ -113,18 +120,20 @@ public void onCreate(Bundle savedInstanceState) {
 
 		conversionManager = new ScratchConversionManager(this, client, false);
 		conversionManager.setCurrentActivity(this);
-		conversionManager.addGlobalDownloadCallback(converterSlidingUpPanelFragment);
-		conversionManager.addBaseInfoViewListener(converterSlidingUpPanelFragment);
-		conversionManager.addGlobalJobViewListener(converterSlidingUpPanelFragment);
-		searchProjectsListFragment.setConversionManager(conversionManager);
+		conversionManager.addGlobalDownloadCallback(getSlidingUpFragment());
+		conversionManager.addBaseInfoViewListener(getSlidingUpFragment());
+		conversionManager.addGlobalJobViewListener(getSlidingUpFragment());
+		getSearchFragment().setConversionManager(conversionManager);
 
-		slidingLayout = (SlidingUpPanelLayout) findViewById(R.id.sliding_layout);
+		slidingLayout = findViewById(R.id.sliding_layout);
 		slidingLayout.addPanelSlideListener(this);
-
-		final int betaLabelColor = ContextCompat.getColor(this, R.color.beta_label_color);
-		appendColoredBetaLabelToTitle(betaLabelColor);
 		hideSlideUpPanelBar();
-		Log.i(TAG, "Scratch Converter Activity created");
+	}
+
+	@Override
+	protected void onPostCreate(Bundle savedBundle) {
+		super.onPostCreate(savedBundle);
+		getSlidingUpFragment().getFinishedFailedJobsAdapter().setScratchProgramOnClickListener(this);
```

Comment:
do you have an answer here?

Reply:
TBH I'm not 100% sure yet myself if it has to be done this way. If we do it at the end of onCreate we crash with a null pointer exception because the finishedJobsAdapter is not yet initialized because that happens within onActivityCreated() of ScratchConverterSlidingUpPanelFragment which gets called after onCreate of ScratchConverterActivity. Do you have a better solution in mind here?